dispatch/static/manager/src/js/containers/AppContainer.js

```diff
@@ -6,13 +6,19 @@ import * as userActions from '../actions/UserActions'
 class App extends React.Component {
 
   componentWillMount() {
-    if (!this.isAuthenticated()) {
-      this.props.onRequireLogin('/')
+    this.checkAuthenticated()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.validToken != this.props.validToken) {
+      if (this.props.validToken === false) {
+        this.props.onRequireLogin('/')
+      }
```

Comment:
I think we can handle this logic directly in the `verifyToken` action with:

```js
dispatch(userActions.requireLogin(nextPath))
```

This would remove the need for the `validToken` store variable (I think)